### DISCUSSION TITLE
Link Pull Request(s) to Work Items

### DISCUSSION THREAD

**Discussion by nicolasproton** [mburumaxwell/paklo#1408] [Ideas]
It would be great if the created Pull Requests (PR) could be automatically linked to Azure DevOps work item(s). All my PRs have automated validation jobs and along with the existing Auto-Complete and Auto-Approve feature, it would enable the dependency update to be automatically merged. The only missing part is the link to Work Items.
For information, I am using TaskV2

Thanks

**Comment by rhyskoedijk** [comment]
It is possible to link to an existing work item, I use this quite frequently with a generic "Update outdated dependencies" work item.
Add `milestone: <work_item_id>` to your dependabot.yml config to enable it, e.g.

```yaml
version: 2
updates:
  - package-ecosystem: "nuget"
    directory: "/"
    milestone: 12345
```

Only one work item can be linked, more documentation here:
https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#milestone

If you are asking if Dependabot could create new work items for each update and link them to the pull request, that isn't supported currently. AFAIK, GitHub Dependabot doesn't do that either, but it sounds like a nice idea.
